ViewModel/LibraryViewModel.swift

```diff
@@ -252,27 +252,28 @@ final class LibraryViewModel: ObservableObject {
         }
     }
 
-    private func fetchData() async throws -> Data? {
+    private func fetchData() async throws -> (Data, URL)? {
         do {
             var request = URLRequest(url: Self.catalogURL, timeoutInterval: 20)
             request.allHTTPHeaderFields = ["If-None-Match": defaults[.libraryETag]]
             let (data, response) = try await self.urlSession.data(for: request)
             guard let response = response as? HTTPURLResponse else { return nil }
             switch response.statusCode {
             case 200:
+                let responseURL = response.url ?? Self.catalogURL
```

Comment:
How could we not get a `response.url`? I think this fallback is asking for trouble.